### ISSUE TITLE
It would be nice if Whisper.file was public

### ISSUE THREAD

**Issue by drawks** [kisielk/whisper-go#8]
I have a carbon listener/cache implemented using whisper-go and at any given point in time there may be many thousand metric which could be updated. I have implemented a LRU to hold the files which prevents a filehandle resource leak by automatically evicting/closing whisper files which have been open but not touched recently. This works for existing whisper files using the OpenWhisper method, since I can open the file and then pass the file object to both the lru and the OpenWhisper method, but when creating a new Whisper object the file is never exposed publicly from the type and no method exists which returns the file.

The only work around I see currently would be to first use Whisper.Create and then Whisper.Close and then reopen the file. This shuffle, while workable seems like it could be factored out with public access to Whisper.file

**Comment by kisielk** [comment]
Yeah, I am not terribly pleased with the API in this package (it was some of the first Go code I wrote years ago). 

Is there any reason your LRU can't contain *whisper.Whisper pointers instead of *os.File though?

**Comment by drawks** [comment]
I suppose it could, but I really like the code being a sort of general lru for file-like objects. It's the classic generic container problem of go.
